File: JavaScript/src/Internals/Functions/DateTimes/Function_HOUR.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';


class Function_HOUR extends Function_1 {
    get Name() {
        return "Hour";
    }

    constructor(a) {
        super(a);
    }

    evaluate(engine, tempParameter) {
        let args1 = this.getDate_1(engine, tempParameter);
        if (args1.IsError) { return args1; }
        return Operand.Create(args1.DateValue.Hour);
    }
}

export { Function_HOUR };

File: csharp/ToolGood.Algorithm/Internals/Functions/DateTimes/Function_HOUR.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.DateTimes
{
	internal sealed class Function_HOUR : Function_1
    {
        public Function_HOUR(FunctionBase func1) : base(func1)
        {
        }

        public override string Name => "Hour";

        public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = GetDate_1(engine, tempParameter);
			if (args1.IsError) { return args1; }
            return Operand.Create(args1.DateValue.Hour);
        }

    }

}
